index.js

```diff
@@ -176,25 +176,29 @@ function redisStore(args) {
    * @param {Function} cb - A callback that returns a potential error and the response
    */
   self.get = function(key, options, cb) {
-    if (typeof options === 'function') {
-      cb = options;
-      options = {};
-    }
-    options = options || {};
-    options.parse = true;
+    return new Promise(function(resolve, reject) {
+      if (typeof options === 'function') {
+        cb = options;
+        options = {};
+      }
+      options = options || {};
+      options.parse = true;
 
-    var compress = (options.compress || options.compress === false) ? options.compress : redisOptions.compress;
-    if (compress) {
-      options.compress = (compress === true) ? compressDefault : compress;
-      key = new Buffer(key);
-    }
+      cb = cb ? cb : (err, result) => err ? reject(err) : resolve(result)
 
-    connect(function(err, conn) {
-      if (err) {
-        return cb && cb(err);
+      var compress = (options.compress || options.compress === false) ? options.compress : redisOptions.compress;
+      if (compress) {
+        options.compress = (compress === true) ? compressDefault : compress;
+        key = new Buffer(key);
       }
 
-      conn.get(key, handleResponse(conn, cb, options));
+      connect(function(err, conn) {
+        if (err) {
+          return cb && cb(err);
```

Comment:
I think that with your latest changes cb will be defined so only cb(err) is enough.